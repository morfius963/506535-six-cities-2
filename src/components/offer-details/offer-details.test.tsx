import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import Map from "../map/map";
import OfferCard from "../offer-card/offer-card";
import OfferDetails from "./offer-details";
import MainHeader from "../main-header/main-header";
import CommentList from "../comments-list/comments-list";
import CommentForm from "../comment-form/comment-form";
import fixtureData, {commentsData} from "../../__fixtures__/offers";

jest.mock(`../map/map`, () => jest.fn().mockReturnValue(null));
jest.mock(`../offer-card/offer-card`, () => jest.fn().mockReturnValue(null));
jest.mock(`../main-header/main-header`, () => jest.fn().mockReturnValue(null));
jest.mock(`../comments-list/comments-list`, () => jest.fn().mockReturnValue(null));
jest.mock(`../comment-form/comment-form`, () => jest.fn().mockReturnValue(null));

describe(`snapshot test`, () => {
  const appData = {
    offers: fixtureData,
    comments: [],
  };
  const user = {
    isAuthorizationRequired: false,
    email: `marf@gmail.com`
  };

  const mockStore = configureStore([]);
  const store = mockStore({
    appData,
    user
  });

  it(`Component correctly renders`, () => {
    const props = {
      location: {
        pathname: `/offer/2`
      },
      email: `Vitalii`,
      onFavoriteCardToggle: jest.fn(),
      onCommentsLoad: jest.fn(),
      onReviewSubmit: jest.fn()
    };

    const tree = renderer
      .create(
            <Provider store={store}>
              <OfferDetails {...props} />
            </Provider>
        )
      .toJSON();

    expect(Map).toHaveBeenCalled();
    expect(OfferCard).toHaveBeenCalled();
    expect(MainHeader).toHaveBeenCalled();
    expect(CommentList).toHaveBeenCalled();
    expect(CommentForm).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });
});
