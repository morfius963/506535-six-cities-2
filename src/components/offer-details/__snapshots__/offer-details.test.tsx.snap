// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Component correctly renders 1`] = `
<div
  className="page"
>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful &amp; luxurious apartment at great location
            </h1>
            <button
              className="property__bookmark-button button "
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height={33}
                width={31}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "20%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              1
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Entire place
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              1
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              3
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              2100
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                1
              </li>
              <li
                className="property__inside-item"
              >
                2
              </li>
              <li
                className="property__inside-item"
              >
                3
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height={74}
                  width={74}
                />
              </div>
              <span
                className="property__user-name"
              >
                Me
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Beautiful &amp; luxurious apartment at great location
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                0
              </span>
            </h2>
          </section>
        </div>
      </div>
      <section
        className="property__map map container"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        />
      </section>
    </div>
  </main>
</div>
`;
